src/main.cpp

```diff
@@ -225,6 +250,56 @@ void dump_document_meta(const std::string filename, PDFDoc *doc,
     packer.pack_nil();
   }
 
+  packer.pack("XFA");
+  if (options.xfa && catalog->getFormType() == Catalog::FormType::XfaForm) {
+    Object xfa;
+    catalog->getAcroForm()->dictLookup("XFA", &xfa);
+
+    if (xfa.isStream()) {
+      pack_stream_content(xfa.getStream());
+    } else {
+      packer.pack_map(xfa.arrayGetLength() / 2);
+      for (int i=0; i<xfa.arrayGetLength()-1; i += 2) {
+        Object key, value;
+        xfa.arrayGet(i, &key);
+        xfa.arrayGet(i+1, &value);
+
+        if (!key.isString() || (!value.isString() && !value.isStream())) {
+          packer.pack_uint32(i / 2);
+          packer.pack_nil();
+          continue;
+        }
+        packer.pack(fmt(&key, uMap));
+        if (value.isString()) {
+          packer.pack(fmt(&value, uMap));
+        } else {
+          pack_stream_content(value.getStream());
+        }
+      }
+    }
+  } else {
+    packer.pack_nil();
```

Comment:
Just an idea: rather than packing `nil` here, would it be cheap to pack some property of the form, for example `xfa.arrayGetLength()` if available else `0`?

Reply:
Done.